File: exercises/101-109_small_problems/easy_01/go/easy_07.go
package main

import "fmt"

func stringy(num int) string {
	i := 0
	str := ""

	for i < num {
		if i%2 == 0 {
			str += "1"
		} else {
			str += "0"
		}

		i++
	}

	return str
}

func main() {
	fmt.Println(stringy(6))
	fmt.Println(stringy(9))
	fmt.Println(stringy(4))
	fmt.Println(stringy(7))
}

File: exercises/101-109_small_problems/easy_01/rust/easy_07.rs
// Stringy Strings
fn main() {
    println!("{}", stringy(6) == "101010");
    println!("{}", stringy(9) == "101010101");
    println!("{}", stringy(4) == "1010");
    println!("{}", stringy(7) == "1010101");
}

fn stringy(l: i32) -> String {
    let mut result = "".to_string();

    for n in 0..l {
        if n % 2 == 0 {
            result = result + "1";
        } else {
            result = result + "0";
        }
    }

    result
}
